tests/stdlib/tvarints.nim

```diff
@@ -0,0 +1,84 @@
+              uint64(high(int32)), uint64(high(int32)), uint64(high(int64))]:
+    let wrLen = writeVu64(dest, test)
+    let rdLen = readVu64(dest, got)
+    doAssert wrLen == rdLen
+    doAssert got == test
+
+  for test in 0u64..300u64:
+    let wrLen = writeVu64(dest, test)
+    let rdLen = readVu64(dest, got)
+    doAssert wrLen == rdLen
+    doAssert got == test
+
+  # check this also works for floats:
+  for test in [0.0, 0.1, 2.0, +Inf, NegInf]:
+    let t = cast[uint64](test)
+    let wrLenB = writeVu64(dest, t)
+    let rdLenB = readVu64(dest, got)
+    doAssert wrLenB == rdLenB
+    doAssert cast[float64](got) == test
+
+block:
+  var hugeIntArray: array[50, byte]
+  var readedInt: uint64
+  doAssert writeVu64(hugeIntArray, 0.uint64) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == 0.uint64
+  doAssert writeVu64(hugeIntArray, uint64.high) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == uint64.high
+  doAssert writeVu64(hugeIntArray, uint64(int64.high)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == uint64(int64.high)
+  doAssert writeVu64(hugeIntArray, uint64(int32.high)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == uint64(int32.high)
+  doAssert writeVu64(hugeIntArray, uint64(int16.high)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == uint64(int16.high)
+  doAssert writeVu64(hugeIntArray, uint64(int8.high)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == uint64(int8.high)
+  doAssert writeVu64(hugeIntArray, cast[uint64](0.0)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](0.0)
+  doAssert writeVu64(hugeIntArray, cast[uint64](-0.0)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](-0.0)
+  doAssert writeVu64(hugeIntArray, cast[uint64](0.1)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](0.1)
+  doAssert writeVu64(hugeIntArray, cast[uint64](0.9555555555555555555555501)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](0.9555555555555555555555501)
+  doAssert writeVu64(hugeIntArray, cast[uint64](+Inf)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](+Inf)
+  doAssert writeVu64(hugeIntArray, cast[uint64](NegInf)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](NegInf)
+  doAssert writeVu64(hugeIntArray, cast[uint64](Nan)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](Nan)
+  doAssert writeVu64(hugeIntArray, cast[uint64](3.1415926535897932384626433)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](3.1415926535897932384626433)
+  doAssert writeVu64(hugeIntArray, cast[uint64](2.71828182845904523536028747)) == readVu64(hugeIntArray, readedInt)
+  doAssert readedInt == cast[uint64](2.71828182845904523536028747)
+
+block:
+  doAssert encodeZigzag(decodeZigzag(0.uint64)) == 0.uint64
```

Comment:
@juancarlospaco good PR; but those tests are not DRY, and hard to read, and should be:
```nim
doAssert encodeZigzag(decodeZigzag(cast[uint64](0.9555555555555555555555501))) == cast[uint64](0.9555555555555555555555501)
```
=>
```nim
template chk(a) =
  doAssert encodeZigzag(decodeZigzag(a)) == a
chk cast[uint64](0.9555555555555555555555501))
```

Reply:
=> fixing in https://github.com/nim-lang/Nim/pull/18089